### ISSUE TITLE
check for measurement bias/error

### ISSUE THREAD

**Issue by ldemailly** [fortio/fortio#3]
Need to check

https://www.youtube.com/watch?v=lJ8ydIuPFeU

http://psy-lob-saw.blogspot.com/2016/07/fixing-co-in-cstress.html

https://github.com/giltene/wrk2/blob/master/README.md

**Comment by ldemailly** [comment]
My understanding is that when fortio reports negative sleep times (ie can't reach the target QPS) one option is to count that falling behind as part of the response time

cc @giltene

I think though the typical methodology may be : 

run fortio -qps 0 to get a rough idea of the max qps (at varying -c )

then run fortio at -qps ~1/2 of max and check latencies

**Comment by zengdgit** [comment]
what's the meaning of sleep time? Does that imply I have sent too many requests?

**Comment by ldemailly** [comment]
yes if you see the warning it means your target system is unable to sustain the requested qps.

**Comment by giltene** [comment]
The issue with a methodology that runs at e.g. 1/2 of the max is that bad latency  behavior is not caused (primarily) by qps. While there is definitely a qps level that will cause terrible latencies, terrible latencies on individual operations can occur even at 1% of the max qps. When the sleep time aheadof any  operations needs to be negative (I.e. the operation should have started in the past) , that sleep time should be added to the observed response time when reporting latency, or coordinated omission will happen. Note that this is not only applicable to the time between two operations, and is supposed to be cumulative. I.e. if you end up “300 operations behind” from a qps perspective, each of the 300 operations should be showing a negative sleep time (as long as the distance from where that operation should have occurred according to the stated qps, and the time.that it was actually started.
